### ISSUE TITLE
[web] New routes are not announced in a11y mode

### ISSUE THREAD

**Issue by nturgut** [flutter/flutter#45207]
The route changes are not communicated correctly to the assistance technology.

**Comment by pedromassangocode** [comment]
Is this still valid today? A little bit of explanation about what the issue it would help a lot.

**Comment by yjbanov** [comment]
This is still an issue. `namesRoute` and `scopesRoute` are not implemented on the web, which is used by dialogs and other routing features. It's not yet clear how to cleanly implement this on the web. I looked at various available ARIA roles. So far I could only think of the following options:

* Use the [dialog](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role) ARIA role. It's not clear if using this role for all routes is a good idea. We could introduce `isDialog` flag to `SemanticsNode` specifically for dialogs to at least cover that use-case.
* Use [live region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) to announce screen changes. We already support live regions on the web, but only if the framework sets the [isLiveRegion](https://api.flutter.dev/flutter/dart-ui/SemanticsFlag/isLiveRegion-constant.html) flag. Perhaps live region is appropriate for routes as well, since routes also fall under "dynamic content which updates without a page reload".
